### PR TITLE
feat: Multiple Reward Locker

### PR DESCRIPTION
**Pull request by piavgh**

### CODE REVIEW

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -301,14 +301,43 @@ export function ButtonError({ error, ...rest }: { error?: boolean } & ButtonProp
...
+  }
+`
+
+export function ButtonDropdown({
+  expanded = true,
+  disabled = false,
+  children,
+  ...rest
+}: { expanded: boolean; disabled?: boolean } & ButtonProps) {
+  const theme = useContext(ThemeContext)
```
Comment (viet-nv) [inline src/components/Button/index.tsx:332]:
```
import useTheme from 'hooks/useTheme'
...
const theme = useTheme()
```
This is better than use useContext. 

Here is some best practices about React Context I read before: https://kentcdodds.com/blog/how-to-use-react-context-effectively

File: src/components/Vesting/RewardLockerSchedules.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+import UnlockedBlock from './UnlockedBlock'
+import { RewardLockerSchedulesWrapper, RewardLockerSchedulesTitle, ClaimAllSection, NoVestingSchedule } from './styleds'
+import { AutoRow, RowBetween } from 'components/Row'
+
+const RewardLockerSchedules = ({
+  rewardLockerAddress,
+  schedules
+}: {
+  rewardLockerAddress: string
+  schedules: any[]
```
Comment (viet-nv) [inline src/components/Vesting/RewardLockerSchedules.tsx:23]: Can we add type for this props?

File: src/components/Vesting/RewardLockerSchedules.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    if (!fullyVestedAlready && !!currentBlockNumber && currentBlockNumber > schedule[1]) {
+      result[address].fullyIndexes.push(schedule[5])
+      result[address].fullyAmount = result[address].fullyAmount.add(BigNumber.from(schedule[2]))
+    }
+
+    result[address].unlockedAmount = result[address].unlockedAmount.add(unlockedAmount)
+    return result
+  }, {})
+
+  const [, setPendingTx] = useState(false)
```
Comment (viet-nv) [inline src/components/Vesting/RewardLockerSchedules.tsx:107]: do we need this state?

File: src/components/Vesting/UnlockedBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import { AutoRow, RowBetween } from 'components/Row'
+import { Reward } from 'state/farms/types'
+import { TYPE } from 'theme'
+import { formattedNum } from 'utils'
+import { useFarmRewardsUSD } from 'utils/dmm'
+import { fixedFormatting } from 'utils/formatBalance'
+import { MenuFlyout, Tag } from './styleds'
+
+const UnlockedBlock = ({ info, onClaimAll }: { info: any; onClaimAll?: () => void }) => {
+  const theme = useContext(ThemeContext)
```
Comment (viet-nv) [inline src/components/Vesting/UnlockedBlock.tsx:17]: same as above. use useTheme is better

File: src/components/Button/index.tsx (hunk 1 above)
Comment (piavgh) [reply src/components/Button/index.tsx:332]: Thanks for the catch, I updated it

File: src/components/Vesting/RewardLockerSchedules.tsx (hunk 3 above)
Comment (piavgh) [reply src/components/Vesting/RewardLockerSchedules.tsx]: I just reused Tien's code.
But now I realized that we can show a TransactionConfirmationModal here, it's much better for UX
Thanks

Comment (piavgh) [reply src/components/Vesting/RewardLockerSchedules.tsx]: I updated this to use a TransactionConfirmationModal

File: src/components/Vesting/RewardLockerSchedules.tsx (hunk 2 above)
Comment (piavgh) [reply src/components/Vesting/RewardLockerSchedules.tsx]: Thanks, using tuple now :D

**Review by viet-nv — APPROVED** [review]

**Pull request merged**